bin/runYANGgenericstats-weekly.sh

```diff
@@ -102,33 +102,38 @@ date +"%c: processing Juniper modules " >> $LOG
 declare -a PIDJUNIPER
 for i in {14..20}
 do
-   if [ $i -eq 15 ]
+   # Juniper/14.2 does not contain subdirectories
+   if [ $i -eq 14 ]
    then
-      continue
-   fi
-   for path in $(ls -d $NONIETFDIR/yangmodels/yang/vendor/juniper/$i*/)
-   do
-      subdircount=`find $path -maxdepth 1 -type d | wc -l`
-      if [ $subdircount -eq 1 ]
-      then
-         echo "path to $path*/ does not exists"
-         continue
-      fi
-      for path2 in $(ls -d $path*/)
-      do
-         git=${path2##*/juniper/}
+         path=$(ls -d $NONIETFDIR/yangmodels/yang/vendor/juniper/$i*/)
+         git=${path##*/juniper/}
          yang_removed=${git%/*}
          prefix=${yang_removed#*/}
          prefix2=$(echo $prefix | tr -cd '[:alnum:]')
-         (python yangGeneric.py --allinclusive True --metadata "JUNIPER $prefix from https://github.com/Juniper/yang/tree/master/$git" --lint True --prefix Juniper$prefix2 --rootdir "$path2" >> $LOG 2>&1) &
-         PIDJUNIPER+=("$!")
+         python yangGeneric.py --allinclusive True --metadata "JUNIPER $prefix from https://github.com/Juniper/yang/tree/master/$git" --lint True --prefix Juniper$prefix2 --rootdir "$path2" >> $LOG 2>&1
+   # Juniper/15* does not exist
+   elif [ $i -eq 15 ]
+   then
+      continue
+   else
+      for path in $(ls -d $NONIETFDIR/yangmodels/yang/vendor/juniper/$i*/)
+      do
+         for path2 in $(ls -d $path*/)
+         do
+            git=${path2##*/juniper/}
+            yang_removed=${git%/*}
+            prefix=${yang_removed#*/}
+            prefix2=$(echo $prefix | tr -cd '[:alnum:]')
+            (python yangGeneric.py --allinclusive True --metadata "JUNIPER $prefix from https://github.com/Juniper/yang/tree/master/$git" --lint True --prefix Juniper$prefix2 --rootdir "$path2" >> $LOG 2>&1) &
+            PIDJUNIPER+=("$!")
+         done
       done
       for PID in $PIDJUNIPER
       do
          wait $PID || exit 1
+         unset PIDJUNIPER
```

Comment:
why is this on line 134 and not after this cycle is done in between line 135 and 136